src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -205,43 +206,57 @@ private RequestDelegate BuildApplication()
 
                 return builder.Build();
             }
-            catch (Exception ex)
+            catch (ReflectionTypeLoadException exception)
```

Comment:
Actually, we check for specific exception types elsewhere: https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.Hosting/Startup/StartupExceptionPage.cs#L54, it would be good to keep it all together.